src/components/themed/TransactionListRow.tsx

```diff
@@ -104,30 +101,43 @@ export function TransactionListRow(props: Props) {
 
   const fiatAmountString = `${fiatSymbol} ${fiatAmount}`
 
-  // Transaction Text and Icon
-  let transactionText, transactionIcon, transactionStyle
-  if (isSentTransaction) {
-    transactionText =
-      action != null
-        ? TX_ACTION_LABEL_MAP[action.type]
-        : transaction.metadata && transaction.metadata.name
-        ? transaction.metadata.name
-        : lstrings.fragment_transaction_list_sent_prefix + selectedCurrencyName
-    transactionIcon = <Ionicons name="arrow-up" size={theme.rem(1.25)} color={theme.negativeText} style={styles.iconArrows} />
-    transactionStyle = styles.iconSent
-  } else {
-    transactionText =
-      action != null
-        ? TX_ACTION_LABEL_MAP[action.type]
-        : transaction.metadata && transaction.metadata.name
-        ? transaction.metadata.name
-        : lstrings.fragment_transaction_list_receive_prefix + selectedCurrencyName
-    transactionIcon = <Ionicons name="arrow-down" size={theme.rem(1.25)} color={theme.positiveText} style={styles.iconArrows} />
-    transactionStyle = styles.iconRequest
-  }
+  // Transaction Title
+  const transactionTitle =
+    action != null
+      ? TX_ACTION_LABEL_MAP[action.type]
+      : transaction.metadata && transaction.metadata.name
+      ? transaction.metadata.name
+      : (isSentTransaction ? lstrings.fragment_transaction_list_sent_prefix : lstrings.fragment_transaction_list_receive_prefix) + selectedCurrencyName
+
+  // Icon & Thumbnail
+  const thumbnailPath = useContactThumbnail(name)
+
+  const isSwapIcon = action != null && (action.type === 'swap' || action.type === 'swapOrderFill')
```

Comment:
Not all currency plugins use the new action system. We should also check `EdgeTransaction.swapData`, since we set that on *all* outgoing swap transactions.